deerlab/snlls.py

```diff
@@ -363,57 +402,52 @@ def ResidualsFcn(p):
     nonlinfit = nonlinfits[globmin]
     sol = sols[globmin]
     Afit = Amodel(nonlinfit)
-    yfit = Afit@linfit
+
+    scales_vec = np.zeros_like(y) 
+    for subset,scale in zip(subsets,scales): 
+        scales_vec[subset] = scale 
+    yfit = scales_vec*(Afit@linfit)
 
     # Uncertainty analysis
-    #--------------------------------------------------------
+    #---------------------
     if uqanalysis:
-        # Compue the residual vector
-        res = weights*(yfit - y)
-
-        # Compute the Jacobian for the linear and non-linear parameters
-        fcn = lambda p: Amodel(p)@linfit
-        Jnonlin = Jacobian(fcn,nonlinfit,lb,ub)
+        # Compute the fit residual
+        res = ResidualsFcn(nonlinfit)
+        
+        # Jacobian (non-linear part)
+        Jnonlin = Jacobian(ResidualsFcn,nonlinfit,lb,ub)
 
-        Jlin = Afit
-        J = np.concatenate((Jnonlin, Jlin),1)
+        # Jacobian (linear part)
+        Jlin = np.zeros((len(res),len(linfit)))
+        Jlin[:len(y),:] = Amodel(nonlinfit)
+        Jlin[len(res)-Nlin:,:] = reg_penalty(regtype, alpha, L, linfit, huberparam, Nnonlin)[1]

```

Comment:
Does this copy the augmented residual into `Jlin`? `reg_penalty` returns the residual first and the Jacobian second, right?

Reply:
Note the `[1]` at the end of the call. It is copying the second output argument, i.e. the Jacobian.